staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -131,6 +131,12 @@ type ClientConfigLoadingRules struct {
 	// WarnIfAllMissing indicates whether the configuration files pointed by KUBECONFIG environment variable are present or not.
 	// In case of missing files, it warns the user about the missing files.
 	WarnIfAllMissing bool
+
+	// WarnIfAllMissingCallback is the warning log callback to use in case of missing files.
+	warnIfAllMissingCallback func(string, ...interface{})
```

Comment:
since all other fields in this struct are exported, why not exporting this too?

Reply:
The reason is strictly related to the other comment: by using the new variable `warningMuted` for distinguish between * `WarnIfAllMissingCallback` is nil and we log by default, and
* `WarnIfAllMissingCallback` is nil and we suppress the log

By letting it exported, it is possible to create a new inconsisent state: `WarnIfAllMissingCallback != nil` and `warningMuted == true`.

For this reason I provided the setter method through which is only possible to have 3 states:
* `WarnIfAllMissingCallback == nil` and `warningMuted == false` (explicit allocation)
*  `WarnIfAllMissingCallback == nil` and `warningMuted == true` (any allocation followed by `MuteWarningMissingList `)
* `WarnIfAllMissingCallback != nil` and `warningMuted == false` (default allocation or explicit allocation followed by a `SetWarnIfAllMissingCallback`)